### PR TITLE
adding some doc comments to `AppModel`

### PR DESCRIPTION
**Pull request by TornaxO7**

### CODE REVIEW

File: src/frontend/window/main_window.rs [hunk 1]
```diff
@@ -20,29 +20,50 @@ pub enum AppInput {
     ScreenshotWindowOutput(ScreenshotWindowOutput),
 }
 
+/// The main struct of `flakeshot` which will manage the
+/// - windows and
+/// - tray
+///
+/// of `flakeshot`.
```
Comment (eneoli) [inline src/frontend/window/main_window.rs:27]: Avoid enumeration of responsibilities. This is subject to change.

```suggestion
/// The main struct of `flakeshot` which will manage the application lifecycle.
```

**Review by eneoli — CHANGES_REQUESTED** [review]

File: src/frontend/window/main_window.rs [hunk 2]
```diff
@@ -20,29 +20,50 @@ pub enum AppInput {
...
 #[derive(Debug)]
 pub struct AppModel {
+    /// Determines how `flakeshot` behaves if it should be closed.
```
[the remark this answers is not in the document]
Comment (TornaxO7) [reply src/frontend/window/main_window.rs]: hm... so what would  you suggest then?

[the remark this answers is not in the document]
Comment (eneoli) [reply src/frontend/window/main_window.rs]: Mode in which flakeshot was started. Or just omit it. `run_mode` speeks for itself.

File: src/frontend/window/main_window.rs [hunk 3]
```diff
@@ -20,29 +20,45 @@ pub enum AppInput {
...
+    fn new(run_mode: RunMode) -> Self {
         AppModel {
             run_mode,
             ui_manager: None,
             window_controllers: vec![],
         }
     }
 
+    /// Start a new GUI session where a screenshot of all monitors
+    /// are taken and opens up the screenshot-editor for you.
```
Comment (eneoli) [inline src/frontend/window/main_window.rs:61]:
```suggestion
    /// are taken and opens up the screenshot-editor.
```

**Review by eneoli — CHANGES_REQUESTED** [review]

**Review by eneoli — APPROVED** [review]
Thank you:)

**Pull request merged**